src/go/rdctl/cmd/reset.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright © 2025 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/spf13/cobra"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/client"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/config"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/factoryreset"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/shutdown"
+)
+
+var (
+	factoryReset bool
+	k8sReset     bool
+)
+
+var resetCmd = &cobra.Command{
+	Use:   "reset",
+	Short: "Reset Rancher Desktop",
+	Long: `Reset Rancher Desktop with various options:
+* Default: Delete the VM and create a new one with current settings
+* --factory: Also delete current settings and show first-run dialog on next start
+* --k8s: Delete only the Kubernetes control plane data. When passed with --factory, the k8s cache will also be cleared`,
```

Comment:
Sorry, I missed this originally that `rdctl factory-reset` has a `--remove-kubernetes-cache` option.

I think it is bad to reset the `--k8s` option for this because it now means different things when used by itself vs. when used with `--factory`. It would only make sense to me if `--k8s` would **always** delete the cache as well, which we don't want.

Although it is quite verbose, I think we should add the `--remove-kubernetes-cache` option to the `reset` command. It could be combined with any of the other options.